src/mistral_common/tokens/tokenizers/instruct.py

```diff
@@ -1010,6 +1044,74 @@ def _encode_tool_calls_in_assistant_message(self, message: AssistantMessageType)
             ]
         return curr_tokens
 
+    def encode_system_message(self, message: SystemMessage) -> List[int]:
+        r"""Encode a system message.
+
+        Args:
+            message: The message to encode.
+
+        Returns:
+            The encoded tokens.
+        """
+        if isinstance(message.content, str):
+            return super().encode_system_message(message)
+
+        tokens = [self.BEGIN_SYSTEM]
+        for chunk in message.content:
+            if isinstance(chunk, TextChunk):
+                tokens += self.tokenizer.encode(chunk.text, bos=False, eos=False)
+            elif isinstance(chunk, ThinkChunk):
+                tokens += self.encode_think(chunk)
+            else:
+                raise ValueError(f"Unsupported chunk type: {type(chunk)}")
+
+        tokens.append(self.END_SYSTEM)
+        return tokens
+
+    def encode_assistant_message(
```

Comment:
too much DUP code. Let's adapt previous encode_assistante_message function so that we only have to add an `encode_think(...)` to the V13 tokenizer